smartsim/settings/arguments/launchArguments.py

```diff
@@ -37,25 +37,32 @@
 logger = get_logger(__name__)
 
 
-class LaunchArgBuilder(ABC):
+class LaunchArguments(ABC):
     """Abstract base class that defines all generic launcher
     argument methods that are not supported.  It is the
     responsibility of child classes for each launcher to translate
     the input parameter to a properly formatted launcher argument.
     """
```

Comment:
Totally agree on a reword, but I actually don't love this sentence:

> It is the responsibility of child classes for each launcher to add methods that translate input parameters to a properly formatted launcher arguments.

Simply because we removed the `finalize` method from this class and made it a standalone fn. The launch args are now more-so responsible for maintaining a consistent state between arguments as a user sets them. Will reword!